ably/rest/auth.py

```diff
@@ -76,13 +76,17 @@ def __init__(self, ably, options):
                              "auth_callback, auth_url, key, token or a TokenDetail")
 
     async def get_auth_transport_param(self):
+        auth_credentials = {}
+        if self.__client_id:
+            auth_credentials["client_id"] = self.__client_id
```

Comment:
this won't actually work because `Auth.client_id` can be set from elsewhere than `Options` (for example, from connection details) and might change over the course of the connection. Since we only care about the client_id from options you can just do:
```suggestion
            auth_credentials["client_id"] = self.auth_options.client_id
```